module-log-parser/src/main/kotlin/org/quiltmc/community/cozy/modules/logs/processors/quilt/FabricModUsedWhenQuiltVersionExistsProcessor.kt

```diff
@@ -0,0 +1,149 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package org.quiltmc.community.cozy.modules.logs.processors.quilt
+
+import dev.kord.core.event.Event
+import io.ktor.client.call.*
+import io.ktor.client.request.*
+import io.ktor.http.*
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonArray
+import kotlinx.serialization.json.decodeFromJsonElement
+import org.quiltmc.community.cozy.modules.logs.Version
+import org.quiltmc.community.cozy.modules.logs.data.LoaderType
+import org.quiltmc.community.cozy.modules.logs.data.Log
+import org.quiltmc.community.cozy.modules.logs.data.Order
+import org.quiltmc.community.cozy.modules.logs.types.LogProcessor
+import java.util.*
+
+private const val MODRINTH_API_BASE = "https://api.modrinth.com/v2"
+
+public class FabricModUsedWhenQuiltVersionExistsProcessor : LogProcessor() {
+
+	override val identifier: String = "fabric-mod-used-when-quilt-version-exists"
+	override val order: Order = Order.Default
+	private val json = Json {
+		ignoreUnknownKeys = true
+	}
+
+	override suspend fun predicate(log: Log, event: Event): Boolean = log.getLoaderVersion(LoaderType.Quilt) != null
+
+	override suspend fun process(log: Log) {
+		val mcVersion = log.minecraftVersion?.string ?: log.getMod("minecraft")?.version?.string ?: return
+
+		val hashToMod = log.getMods().filter {
+			it.value.type?.lowercase(Locale.getDefault()).equals("fabric") && it.value.hash != null
+		}.map { (_, mod) -> mod.hash!! to mod }.toMap()
+		if (hashToMod.isEmpty()) return
+
+		val hashesRequest = client.post {
+			url("$MODRINTH_API_BASE/version_files")
+			setBody(json.encodeToString(HashLookupParam(hashToMod.keys.toList(), "sha1")))
+			contentType(ContentType.Application.Json)
+		}
```

Comment:
Odd mismatch from the GET requests
```suggestion
		val hashesRequest = client.post("$MODRINTH_API_BASE/version_files") {
			setBody(json.encodeToString(HashLookupParam(hashToMod.keys.toList(), "sha1")))
			contentType(ContentType.Application.Json)
		}
```